src/overlays/actors/ovl_En_Pr2/z_en_pr2.c

```diff
@@ -71,48 +92,674 @@ const ActorInit En_Pr2_InitVars = {
+            func_80A751B4(this);
+        }
+    }
+
+    if (this->collider.base.atFlags & AT_BOUNCED) {
+        this->actor.speedXZ = -10.0f;
+    }
+}
+
+void EnPr2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnPr2* this = THIS;
+    f32 rand;
+
+    if (thisx) {}
+
+    Actor_SetScale(&this->actor, this->unk_204);
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->unk_1DA != 0) {
+        this->unk_1DA--;
+    }
+
+    if (this->unk_1D8 != 0) {
+        this->unk_1D8--;
+    }
 
-extern DamageTable D_80A75BC0;
-extern ColliderCylinderInit D_80A75BE0;
+    if (this->unk_1DC != 0) {
+        this->unk_1DC--;
+    }
 
-extern UNK_TYPE D_06003904;
-extern UNK_TYPE D_06004340;
+    if (this->unk_1DE != 0) {
+        this->unk_1DE--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/EnPr2_Init.s")
+    Actor_SetFocus(&this->actor, 10.0f);
+    func_80A755D8(this, globalCtx);
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0, 10.0f, 20.0f, 0x1F);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/EnPr2_Destroy.s")
+    if (this->unk_1D6) {
+        s32 i;
+        Vec3f sp58;
+        f32 rand;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A7429C.s")
+        Math_Vec3f_Copy(&sp58, &this->unk_270);
+        this->unk_1D6 = false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A7436C.s")
+        sp58.x += randPlusMinusPoint5Scaled(20.0f);
+        sp58.y += randPlusMinusPoint5Scaled(5.0f);
+        sp58.z += randPlusMinusPoint5Scaled(20.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A74510.s")
+        for (i = 0; i < 2; i++) {
+            rand = Rand_ZeroFloat(0.03f) + 0.07f;
+            EffectSsBubble_Spawn(globalCtx, &sp58, 0, 5.0f, 5.0f, rand);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A745C4.s")
+    if ((this->unk_1F4 == 255) && (this->unk_1E0 < 10)) {
+        this->actor.shape.rot.x = this->actor.world.rot.x;
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        this->actor.shape.rot.z = this->actor.world.rot.z;
+        if (this->unk_1D4 != 3) {
+            Collider_UpdateCylinder(&this->actor, &this->collider);
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A745FC.s")
+s32 func_80A758E8(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnPr2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pr2/func_80A74888.s")
+    if (this->unk_1E0 < 10) {
+        if (limbIndex == 2) {
+            rot->y += (s16)this->unk_1FC * -1;
```

Comment:
Is the explicit `* -1` required?
Could it be something like?
```c
rot->y += (s16)-this->unk_1FC;
```
or?
```c
rot->y += -(s16)this->unk_1FC;
```
or?
```c
rot->y -= (s16)this->unk_1FC;
```

Reply:
None of those work unfortunately, and none others I tried. Some get close, but not to match.